code/modules/mob/living/carbon/human/human_attackhand.dm

```diff
@@ -59,6 +59,17 @@
 
 	switch(M.a_intent)
 		if(I_HELP)
+			if(can_operate(src))
+				if(health >= config.health_threshold_crit)
+					if(src.surgeries.len)
+						for(var/datum/surgery/S in src.surgeries)
+							if(istype(S.get_surgery_step(), /datum/surgery_step/cavity/place_item) || istype(S.get_surgery_step(),/datum/surgery_step/remove_object))

```

Comment:
@Aurorablade that's my point, if the next step doesn't need a hand, it'll return 0, so you execute help_shake_act, otherwise it'll stop